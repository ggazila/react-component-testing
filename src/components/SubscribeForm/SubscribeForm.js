import React, { useState } from 'react';
import "./SubscribeForm.css";

const SubscribeForm = () => {

    const [isDisabled, setIsDisabled] = useState(true);
    const [email, setEmail] = useState("");

    function handleChange(e){
        setEmail(e.target.value);
        setIsDisabled(e.target.value === "");
    }

    return (
        <div className="container">
            <h1>Subscribe To Their Newsletter</h1>
            <form className="form">
                <label htmlFor="email">Email Address</label>
                <input onChange={handleChange} type="email" id="email" name="email" placeholder="Email Address" value={email} />
                <input type="checkbox" name="agreement_checkbox" id="agreement_checkbox" />
                <label htmlFor="agreement_checkbox">I agree to disagree whatever the terms and conditions are.</label>
                <button name="subscribe-button" type="submit" className="button" disabled={isDisabled} >Subscribe</button>
            </form>
        </div>
    );
};

export default SubscribeForm;
