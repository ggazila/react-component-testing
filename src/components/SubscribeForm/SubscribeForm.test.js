import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SubscribeForm from './SubscribeForm';
import renderer from 'react-test-renderer';

it("The subscribe button is disabled before typing anything in the input text box", () => {
  render(<SubscribeForm />);

  expect(screen.getByRole("button", { name: /subscribe/i })).toBeDisabled();
});

it("The subscribe button becomes enabled when we start typing in the input text box", () => {
  render(<SubscribeForm />);

  userEvent.type(screen.getByRole("textbox", { name: /email/i }), "abc@email.com");

  expect(screen.getByRole("button", { name: /subscribe/i })).toBeEnabled();
});

it("Test to match snapshot of component", () => {
  const subscribeFormTree = renderer.create(<SubscribeForm />).toJSON();
  expect(subscribeFormTree).toMatchSnapshot();
})
