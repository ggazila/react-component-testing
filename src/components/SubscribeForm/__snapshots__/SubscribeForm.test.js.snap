// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to match snapshot of component 1`] = `
<div
  className="container"
>
  <h1>
    Subscribe To Their Newsletter
  </h1>
  <form
    className="form"
  >
    <label
      htmlFor="email"
    >
      Email Address
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="email"
      value=""
    />
    <input
      id="agreement_checkbox"
      name="agreement_checkbox"
      type="checkbox"
    />
    <label
      htmlFor="agreement_checkbox"
    >
      I agree to disagree whatever the terms and conditions are.
    </label>
    <button
      className="button"
      disabled={true}
      name="subscribe-button"
      type="submit"
    >
      Subscribe
    </button>
  </form>
</div>
`;
